src/qililab/platform/platform.py

```diff
@@ -898,29 +906,51 @@ def _execute_qblox_compilation_output(self, output: QbloxCompilationOutput, debu
         return results
 
     def _execute_quantum_machines_compilation_output(
-        self, output: QuantumMachinesCompilationOutput, cluster: QuantumMachinesCluster, debug: bool = False
+        self,
+        output: QuantumMachinesCompilationOutput,
+        cluster: QuantumMachinesCluster,
+        timeout_tries: int = 3,
+        debug: bool = False,
     ):
         qua, configuration, measurements = output.qua, output.configuration, output.measurements
-        cluster.append_configuration(configuration=configuration)
-
-        if debug:
-            with open("debug_qm_execution.py", "w", encoding="utf-8") as sourceFile:
-                print(generate_qua_script(qua, cluster.config), file=sourceFile)
-
-        compiled_program_id = cluster.compile(program=qua)
-        job = cluster.run_compiled_program(compiled_program_id=compiled_program_id)
-
-        acquisitions = cluster.get_acquisitions(job=job)
-
-        results = QProgramResults()
-        # Doing manual classification of results as QM does not return thresholded values like Qblox
-        for measurement in measurements:
-            measurement_result = QuantumMachinesMeasurementResult(
-                measurement.bus,
-                *[acquisitions[handle] for handle in measurement.result_handles],
-            )
-            measurement_result.set_classification_threshold(measurement.threshold)
-            results.append_result(bus=measurement.bus, result=measurement_result)
+        for iteration in np.arange(timeout_tries):  # TODO: This is a temporal fix as QM fixes the dataloss error
```

Comment:
the comment indicates this is for dataloss error but it is not, this is for the infinite hanging up, no?

Reply:
yes, the name was from a previous version, now it says timeout instead